coresimd/acle/mod.rs

```diff
@@ -0,0 +1,140 @@
+//! ARM C Language Extensions (ACLE)
+//!
+//! # Developer notes
+//!
+//! Below is a list of built-in targets that are representative of the different ARM
+//! architectures; the list includes the `target_feature`s they possess.
+//!
+//! - `armv4t-unknown-linux-gnueabi` - **ARMv4** - `+v4t`
+//! - `armv5te-unknown-linux-gnueabi` - **ARMv5TE** - `+v4t +v5te`
```

Comment:
`rustc --print cfg` says otherwise

``` console
$ rustc --print cfg --target armv5te-unknown-linux-gnueabi | grep feature
target_feature="v5te"

$ rustc --print cfg --target thumbv7em-none-eabi | grep feature
target_feature="dsp"
target_feature="mclass"
target_feature="v5te"
target_feature="v6"
target_feature="v6k"
target_feature="v6t2"
target_feature="v7"
```

All the info written here corresponds to what `rustc --print cfg` outputs

Reply:
Hmm. LLVM's definition of "dsp" doesn't seem to match ACLE's definition. From my point of view this looks like an LLVM bug. Let's just work around it for now by using `(v5te && !mclass) || dsp`.

> __ARM_FEATURE_DSP is defined to 1 if the DSP (v5E) instructions are supported and the intrinsics defined in 9.4 are available. These instructions include QADD, SMULBB etc. This feature also implies support for the Q flag.